src/Canvas.tsx

```diff
@@ -108,7 +108,7 @@ function Canvas<R, K extends keyof R>({
       const isCellAtLeftBoundary = left < scrollLeft + width + getFrozenColumnsWidth();
       const isCellAtRightBoundary = left + width > clientWidth + scrollLeft;
       if (isCellAtLeftBoundary || isCellAtRightBoundary) {
-        const newScrollLeft = getColumnScrollPosition(columns, idx, scrollLeft, clientWidth);
+        const newScrollLeft = getColumnScrollPosition(columns as readonly CalculatedColumn<R, never>[], idx, scrollLeft, clientWidth);
```

Comment:
Please avoid type assertions, we can fix `getColumnScrollPosition` by adding `SR` generic.
```suggestion
        const newScrollLeft = getColumnScrollPosition(columns, idx, scrollLeft, clientWidth);
```